src/fluree/db/query/fql/syntax.cljc

```diff
@@ -117,9 +123,10 @@
                              [:desc [:fn desc?]]]
      ::ordering             [:orn
                              [:scalar ::var]
-                             [:vector [:catn
-                                       [:direction ::direction]
-                                       [:dimension ::var]]]]
+                             [:vector [:and list?
```

Comment:
I'm surprised to see `list?` here. Is that what we want internally? I.e. `(:asc :foo/bar)`?

Reply:
Yeah, it's essential that it be a list internally because of the way the query engine recognizes these clauses. 

You can have a single grouping in your group-by clause, or multiple groupings wrapped in a vector. 

An individual grouping is either a variable or a list containing a direction and a variable. 

We need to tell the difference between a single grouping that has both a direction and variable, and a collection of multiple groupings, so we use a list to represent the single grouping with direction and a vector to represent multiple groupings. 

This was the quickest way I could think of to encode these two cases without too many extra steps traversing into nested vectors and inspecting what they are, but we can change it if there's a better way.

Reply:
Makes sense, and I think this is fine.